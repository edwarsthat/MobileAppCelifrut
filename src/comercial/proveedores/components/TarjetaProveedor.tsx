import React from "react";
import { Text, View, StyleSheet } from "react-native";
import { proveedoresType } from "../../../../types/proveedoresType";

type propsType = {
    proveedor: proveedoresType
}

export default function TarjetaProveedor(props: propsType): React.JSX.Element {
    return (
        <View style={styles.cardContainer}>
            {/* Encabezado principal */}
            <Text style={styles.headerText}>
                Codigo {props.proveedor['CODIGO INTERNO'] ?? ''}
            </Text>

            {/* Información general */}
            <View style={styles.infoRow}>
                <Text style={styles.label}>Predio:</Text>
                <Text style={styles.value}>{props.proveedor.PREDIO ?? ''}</Text>
            </View>

            {/* Sección ICA */}
            <View style={styles.section}>
                <Text style={styles.sectionTitle}>ICA</Text>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Código:</Text>
                    <Text style={styles.value}>{props.proveedor.ICA && props.proveedor.ICA.code}</Text>
                </View>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Tipo de Fruta:</Text>
                    <Text style={styles.value}>{
                        props.proveedor.ICA &&
                        props.proveedor.ICA.tipo_fruta &&
                        props.proveedor.ICA.tipo_fruta.reduce((acu, item) => (acu += item + ', '), '')
                    }</Text>
                </View>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Vencimiento:</Text>
                    <Text style={styles.value}>{
                    props.proveedor.ICA && props.proveedor.ICA.fechaVencimiento}</Text>
                </View>
            </View>

            {/* Sección GGN */}
            <View style={styles.section}>
                <Text style={styles.sectionTitle}>GGN</Text>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Código:</Text>
                    <Text style={styles.value}>{props.proveedor?.GGN?.code ?? ''}</Text>
                </View>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Tipo de Fruta:</Text>
                    <Text style={styles.value}>{
                        props.proveedor.GGN &&
                        props.proveedor.GGN.tipo_fruta &&
                        props.proveedor.GGN.tipo_fruta.reduce((acu, item) => (acu += item + ', '), '')
                    }</Text>
                </View>
                <View style={styles.infoRow}>
                    <Text style={styles.label}>Vencimiento:</Text>
                    <Text style={styles.value}>{props.proveedor.GGN && props.proveedor.GGN.fechaVencimiento }</Text>
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    cardContainer: {
        backgroundColor: '#FFFFFF',     // Fondo de tarjeta
        marginHorizontal: 16,          // Separación horizontal
        marginVertical: 8,             // Separación vertical
        padding: 16,                   // Espacio interno
        borderRadius: 8,               // Esquinas redondeadas
        // Sombra en iOS
        shadowColor: '#000',
        shadowOffset: { width: 0, height: 2 },
        shadowOpacity: 0.1,
        shadowRadius: 4,
        // Sombra en Android
        elevation: 2,
    },
    headerText: {
        fontSize: 18,
        fontWeight: 'bold',
        marginBottom: 8,
        color: '#333333',
    },
    infoRow: {
        flexDirection: 'row',          // Para ubicar label y valor en línea
        marginBottom: 4,
    },
    label: {
        fontWeight: '600',
        color: '#555555',
        marginRight: 4,
    },
    value: {
        color: '#777777',
    },
    section: {
        marginTop: 12,                 // Separación superior
        paddingTop: 8,                 // Padding interno superior
        borderTopWidth: 1,             // Línea para separar secciones
        borderTopColor: '#EEEEEE',
    },
    sectionTitle: {
        fontSize: 16,
        fontWeight: 'bold',
        marginBottom: 8,
        color: '#333333',
    },
});
