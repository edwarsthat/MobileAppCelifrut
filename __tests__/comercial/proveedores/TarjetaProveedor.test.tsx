import { describe, test, expect } from '@jest/globals';
import TarjetaProveedor from '../../../src/comercial/proveedores/components/TarjetaProveedor';
import renderer from 'react-test-renderer';
import React from 'react';
import { render } from '@testing-library/react-native';

describe('TarjetaProveedor Component', () => {
    const mockProveedor = {
        'CODIGO INTERNO': 12345,
        PREDIO: 'Finca La Esperanza',
        ICA: {
            code: 'ICA-6789',
            tipo_fruta: ['Banana', 'Mango'],
            fechaVencimiento: '2025-12-31',
        },
        GGN: {
            code: 'GGN-9876',
            tipo_fruta: ['Banana', 'Papaya'],
            fechaVencimiento: '2026-06-15',
            paises: ['Colombia', 'Ecuador'],
        },
    };

    test('Debe renderizar correctamente con datos v치lidos', () => {
        const tree = renderer.create(<TarjetaProveedor proveedor={mockProveedor} />).toJSON();
        expect(tree).toMatchSnapshot(); // Compara la UI renderizada con una versi칩n guardada
    });

    test('Debe renderizar correctamente cuando faltan datos', () => {
        const proveedorIncompleto = {
            'CODIGO INTERNO': null,
            PREDIO: '',
            ICA: {},
            GGN: null,
        };
        const tree = renderer.create(<TarjetaProveedor proveedor={proveedorIncompleto} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('Debe renderizar correctamente cuando faltan datos 2', () => {
        const proveedorIncompleto = {
            'CODIGO INTERNO': undefined,
            PREDIO: null,
            ICA: null,
            GGN: {},
        };
        const tree = renderer.create(<TarjetaProveedor proveedor={proveedorIncompleto} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Debe renderizar el c칩digo del proveedor', () => {
        const { getByText } = render(<TarjetaProveedor proveedor={mockProveedor} />);
        expect(getByText(/Codigo 12345/i)).toBeTruthy();
    });

    test('Debe renderizar el predio correctamente', () => {
        const { getByText } = render(<TarjetaProveedor proveedor={mockProveedor} />);
        expect(getByText(/Finca La Esperanza/i)).toBeTruthy();
    });
    test('Debe renderizar los datos de ICA correctamente', () => {
        const { getByText } = render(<TarjetaProveedor proveedor={mockProveedor} />);
        expect(getByText(/ICA-6789/i)).toBeTruthy();
        expect(getByText(/Banana, Mango,/i)).toBeTruthy();
        expect(getByText(/2025-12-31/i)).toBeTruthy();
    });

    // test('Debe renderizar los datos de GGN correctamente', () => {
    //     const { getByText } = render(<TarjetaProveedor proveedor={mockProveedor} />);
    //     expect(getByText(/GGN/i)).toBeTruthy();
    //     expect(getByText(/C칩digo:/i)).toBeTruthy();
    //     expect(getByText(/GGN-9876/i)).toBeTruthy();
    //     expect(getByText(/Tipo de Fruta:/i)).toBeTruthy();
    //     expect(getByText(/Banana, Papaya/i)).toBeTruthy();
    //     expect(getByText(/Vencimiento:/i)).toBeTruthy();
    //     expect(getByText(/2026-06-15/i)).toBeTruthy();
    // });
});
