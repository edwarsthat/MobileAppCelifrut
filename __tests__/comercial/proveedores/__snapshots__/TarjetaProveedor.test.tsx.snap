// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TarjetaProveedor Component Debe renderizar correctamente con datos válidos 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 2,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#333333",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Codigo 
    12345
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#555555",
          "fontWeight": "600",
          "marginRight": 4,
        }
      }
    >
      Predio:
    </Text>
    <Text
      style={
        {
          "color": "#777777",
        }
      }
    >
      Finca La Esperanza
    </Text>
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      ICA
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        ICA-6789
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        Banana, Mango, 
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        2025-12-31
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      GGN
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        GGN-9876
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        Banana, Papaya, 
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      >
        2026-06-15
      </Text>
    </View>
  </View>
</View>
`;

exports[`TarjetaProveedor Component Debe renderizar correctamente cuando faltan datos 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 2,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#333333",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Codigo 
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#555555",
          "fontWeight": "600",
          "marginRight": 4,
        }
      }
    >
      Predio:
    </Text>
    <Text
      style={
        {
          "color": "#777777",
        }
      }
    />
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      ICA
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      GGN
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`TarjetaProveedor Component Debe renderizar correctamente cuando faltan datos 2 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "elevation": 2,
      "marginHorizontal": 16,
      "marginVertical": 8,
      "padding": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#333333",
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Codigo 
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 4,
      }
    }
  >
    <Text
      style={
        {
          "color": "#555555",
          "fontWeight": "600",
          "marginRight": 4,
        }
      }
    >
      Predio:
    </Text>
    <Text
      style={
        {
          "color": "#777777",
        }
      }
    />
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      ICA
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#EEEEEE",
        "borderTopWidth": 1,
        "marginTop": 12,
        "paddingTop": 8,
      }
    }
  >
    <Text
      style={
        {
          "color": "#333333",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
    >
      GGN
    </Text>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Código:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Tipo de Fruta:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
    <View
      style={
        {
          "flexDirection": "row",
          "marginBottom": 4,
        }
      }
    >
      <Text
        style={
          {
            "color": "#555555",
            "fontWeight": "600",
            "marginRight": 4,
          }
        }
      >
        Vencimiento:
      </Text>
      <Text
        style={
          {
            "color": "#777777",
          }
        }
      />
    </View>
  </View>
</View>
`;
